quantus/metrics/robustness/avg_sensitivity.py

```diff
@@ -6,26 +6,44 @@
 # You should have received a copy of the GNU Lesser General Public License along with Quantus. If not, see <https://www.gnu.org/licenses/>.
 # Quantus project URL: <https://github.com/understandable-machine-intelligence-lab/Quantus>.
 
+import sys
 from typing import Any, Callable, Dict, List, Optional
+
 import numpy as np
 
-from quantus.helpers import asserts
 from quantus.functions import norm_func
-from quantus.helpers import warn
-from quantus.helpers.model.model_interface import ModelInterface
 from quantus.functions.normalise_func import normalise_by_max
-from quantus.functions.perturb_func import uniform_noise, perturb_batch
+from quantus.functions.perturb_func import perturb_batch, uniform_noise
 from quantus.functions.similarity_func import difference
-from quantus.metrics.base_batched import BatchedPerturbationMetric
+from quantus.helpers import asserts, warn
 from quantus.helpers.enums import (
-    ModelType,
     DataType,
-    ScoreDirection,
     EvaluationCategory,
+    ModelType,
+    ScoreDirection,
+)
+from quantus.helpers.model.model_interface import ModelInterface
+from quantus.helpers.perturbation_utils import (
+    make_changed_prediction_indices_func,
+    make_perturb_func,
 )
+from quantus.metrics.base import Metric
+
+# This file is part of Quantus.
+# Quantus is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+# Quantus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public License along with Quantus. If not, see <https://www.gnu.org/licenses/>.
+# Quantus project URL: <https://github.com/understandable-machine-intelligence-lab/Quantus>.
+
+
+if sys.version_info >= (3, 8):
```

Comment:
@aaarrti

Reply:
`@final` shows users, that this class must not be inherited from.
E.g.,

```python
from typing import final


@final
class A:
    pass


class B(A):
    pass
```

```shell
mypy example.py 

mypy.ini: [mypy]: Unrecognized option: show_none_errors = False
example.py:9: error: Cannot inherit from final class "A"  [misc]
Found 1 error in 1 file (checked 1 source file)
```
The `if sys.version_info >= (3, 8)` can be removed, after we drop python3.7 support (which I hope will be soon, because it is not maintained since half a year https://devguide.python.org/versions/)